Homework/Le_Trong/src/lesson_06/max_area_of_island.rs

```diff
@@ -0,0 +1,78 @@
+struct Grid<'a> {
+    matrix: &'a Vec<Vec<i32>>,
+    rows: usize,
+    cols: usize,
+    visited: Vec<Vec<bool>>,
+    dr: [isize; 4],
+    dc: [isize; 4],
+}
+
+impl<'a> Grid<'a> {
+    fn new(matrix: &'a Vec<Vec<i32>>) -> Self {
+        return Grid {
+            matrix,
+            rows: matrix.len(),
+            cols: matrix[0].len(),
+            visited: vec![vec![false; matrix[0].len()]; matrix.len()],
+            dr: [0, 1, 0, -1],
+            dc: [1, 0, -1, 0],
+        };
+    }
+
+    fn count_lands_aka_bfs(&mut self, r: usize, c: usize) -> u32 {
```

Comment:
dfs, not bfs

Reply:
Yup, it goes down to the tree first. It's dfs.